### DISCUSSION TITLE
math: kerning is wrong

### DISCUSSION THREAD

**Discussion by nschloe** [community/community#16997] [Other Feature Feedback, Questions, & Ideas] [answered]
Check out the kerning in `a=b`,

$$
a = b
$$

GitHub:
![kerning-github](https://user-images.githubusercontent.com/181628/169559313-37591711-fc05-40cf-bcf0-f08219103a0a.png)

LaTeX:
![kerning-latex](https://user-images.githubusercontent.com/181628/169559283-fd9a9bd5-0d32-4d6f-9e39-94f7b6f8c4cb.png)

Likewise, almost all other spacings are wrong.

**Comment by Simran-B** [comment]
This might be yet another case of GitHub processing the input in the wrong order, Markdown before math. Escaping spaces seems to help:

`$a\ =\ b$`

$a\ =\ b$

**Reply by nschloe** [reply]
I'd rather think it's a font misconfiguration in their MathJax, not an order issue.

**Reply by UlyssesZh** [reply]
No, it is not such an order problem. In fact, in LaTeX, even if you do not include the spaces around the `=` sign, the kernings are the same as if there were spaces, which is correct. Actually, MathJax usually have correct kernings, so there should be something wrong with GitHub's configuration with MathJax in the webpages.

**Comment by UlyssesZh** [comment]
This is related to #16939, which is also an issue related to the order of rendering Markdown and rendering LaTeX.
Currently, you can only use the workaround:
```md
$a\\,=\\,b$
```
$a\\,=\\,b$
Note that here double slashes are required (refer to #17143).

**Comment by nschloe** [comment, marked as answer]
This is fixed (for) now.

**Reply by UlyssesZh** [reply]
Don't mark as answer because it is still not fixed.
GitHub used a stupid workaround: it just added some spacing between any characters!
Now letters that should not have space between have extra kerning, e.g. those in \operatorname or \mathit.

**Reply by nschloe** [reply]
Oh dear. Can you post an example with screenshot?

**Comment by UlyssesZh** [comment]
> Can you post an example with screenshot?

@nschloe It seems that it is a browser-specific problem...
```latex
$$\mathit{xyz}\operatorname{sgn}\sin xyz$$
```
Rendered by pdfLaTeX (with `\usepackage{amsmath}`):
![image](https://user-images.githubusercontent.com/26196187/170645849-d30940ab-fdf0-433a-b3d9-f03ed9eb5a8c.png)
Rendered by GitHub:
$$\mathit{xyz}\operatorname{sgn}\sin xyz$$
On Microsoft Edge on Windows (largely expected):
![image](https://user-images.githubusercontent.com/26196187/170646245-a1933ae1-3b7a-4672-85f2-f5dd03175feb.png)
While on Microsoft Edge on Android mobile (too much spacing) (switching to desktop view does not help):
![image](https://user-images.githubusercontent.com/26196187/170646838-e0502368-18c4-43d2-905b-64718f2a7cbd.png)

**Reply by UlyssesZh** [reply]
After some playing around, I found that this should be a bug of MathJax (which I do not remember it had some time before). I will do more research about that.

**Reply by UlyssesZh** [reply]
I didn't find anything on the internet. I posted [an issue](https://github.com/mathjax/MathJax/issues/2877) in MathJax.
